### DISCUSSION TITLE
repeat execution of 'gio --version' during execsnoop.bt

### DISCUSSION THREAD

**Discussion by JLT032** [bpftrace/bpftrace#2099] [Q&A] [answered]
bpfn00b here, while running `sudo execsnoop.bt` i noticed continuous 'gio --version'

i assume this is somehow a consequence of executing bpftrace ?
i assume this based on  attempting to track the PID but i could not find it back using `ps -ef | grep <PID>`

**Comment by fbs** [comment, marked as answer]
These are things your system does, bpftrace does not depend on external binaries not does it execute them, unless you use the `system()` call from within your program.

As for `gio` a quick google shows  https://man.archlinux.org/man/core/glib2/gio.1.en  and 

Package name:
    core/glib2
Version:
    2.70.2-1
Upstream:
    https://wiki.gnome.org/Projects/GLib

So it seems related to your glib / gnome (window manager)  setup.

**Reply by JLT032** [reply]
Thanks, sorry for the waste of time, i woke up way too early and assumed to do something productive. Learning about BPFtrace was okay, asking questions though, not so much.

**Reply by fbs** [reply]
Don't worry, it's a valid question, not a waste of time :) . 

Systems spawn a lot of helper tools which you might not be familiar with. Best to check the man/help or search for `man <tool>` or `linux <tool>` and see what they are used for. 

` i assume this based on attempting to track the PID but i could not find it back using ps -ef | grep <PID>`

for short lived processes you can make a execsnoop++ bpftrace program that traces `fork`s as well as `execve`s so you know which process spawned it
